### DISCUSSION TITLE
How to get socket remote address from loader request

### DISCUSSION THREAD

**Discussion by z1chir** [remix-run/remix#2713] [Q&A] [answered]
How to get socket remote address from loader request

**Comment by kiliman** [comment, marked as answer]
That's not really a Remix question. You need to get that from your platform provider. Typically they would set a header like `X-Client-IP` or `X-Forwarded-For`.

Here's a helper function that checks several different headers.

https://github.com/sergiodxa/remix-utils#getclientipaddress

**Reply by retrievaaaa** [reply]
There should still be a way to get the actual IP address connecting, something like request.socket.remoteAddress, but I cant find anything like that.
